File: backend/src/domain/repositories/file.py
from abc import ABC, abstractmethod
from typing import Optional
from src.domain.entities.models import SessionFile


class IFileRepository(ABC):
    """会话文件数据仓储接口"""

    @abstractmethod
    async def upsert(
        self,
        session_id: str,
        user_id: str,
        file_path: str,
        file_name: str,
        file_size: int,
        mime_type: str,
        category: str,
        storage_type: str = "sandbox",
        task_id: Optional[str] = None,
        storage_key: Optional[str] = None,
    ) -> SessionFile:
        """创建或根据 (session_id, file_path) 覆盖更新文件记录"""
        ...

    @abstractmethod
    async def list_by_session(
        self,
        session_id: str,
        category: Optional[str] = None,
        limit: int = 50,
        offset: int = 0,
    ) -> tuple[list[SessionFile], int]:
        """查询指定会话的文件列表（返回列表及总数）"""
        ...

    @abstractmethod
    async def get_by_id(self, file_id: str) -> Optional[SessionFile]:
        """根据文件 ID 查询单个文件信息"""
        ...

    @abstractmethod
    async def delete_by_id(self, file_id: str) -> bool:
        """标记删除或物理删除文件"""
        ...

File: frontend/src/types/file.ts
export type FileCategory = "all" | "html" | "image" | "code" | "document" | "data";

export interface SessionFile {
  id: string;
  session_id: string;
  user_id: string;
  task_id?: string | null;
  file_name: string;
  file_path: string;
  file_size: number;
  mime_type: string;
  category: "html" | "image" | "code" | "document" | "data" | string;
  storage_type: string;
  preview_url: string;
  download_url: string;
  created_at: string;
  updated_at: string;
}

export interface FileListResponse {
  total: number;
  items: SessionFile[];
}
